### ISSUE TITLE
How to access Engine::Mesh in plugins

### ISSUE THREAD

**Issue by nmellado** [STORM-IRIT/Radium-Engine#244]
In Radium's main app, it is possible to access loaded geometry in plugins by overriding the method `System::handleAssetLoading`, and from the input entity, get access to the `TriangleMesh`:

```c++
// get entity
auto em =  Ra::Engine::RadiumEngine::getInstance()->getEntityManager();
Ra::Engine::Entity* e = em->entityExists(entityName) ?
                        em->getEntity(entityName):
                        nullptr;
// get TriangleMesh provided by FancyMesh
bool hasMesh = ComponentMessenger::getInstance()->canRw<TriangleMesh>( e, id );

if (hasMesh){
  mesh = ComponentMessenger::getInstance()->rwCallback<TriangleMesh>( e, id )();
}
```

Using the same mechanism, is it possible to access the `Ra::Engine::Mesh` that is associated to an entity, if it exists ?

**Comment by louen** [comment]
Basically you have to navigate the hierarchy Entity -> Component -> Render Object -> Mesh
Generally speaking,  you need to know the Entity name, the component name and the RO name (or its number in the Component).
```c++
// Input variables 
std:string entityName = ...
std::string componentName = ...
std::string roName = ...

// 1) Get entity
auto em =  Ra::Engine::RadiumEngine::getInstance()->getEntityManager();
Ra::Engine::Entity* e = em->entityExists(entityName) ?
                        em->getEntity(entityName):
                        nullptr;

// 2) Get component
const Ra::Engine::Component* c = e->getComponent(componentName);

// 3) Get RO
const Ra::Engine::RenderObjectManager* roMgr =
    Ra::Engine::RadiumEngine::getInstance()->getRenderObjectManager();
for (const auto& idx : c->m_renderObjects) {
    if (roMgr->getRenderObject(idx)->getName() ==  roName) {
        return roMgr->getRenderObject(idx)->getMesh(); // Here's your mesh
    }
}
```

**Comment by nmellado** [comment]
Ok thanks.
Taking inspiration from your code, I've added a function in the Radium Engine Renderer (176b9f3, see PR #243 ):

```c++
        Mesh *RadiumEngine::getMesh(const std::string &entityName,
                                    const std::string &componentName,
                                    const std::string &roName) const
        {

            // 1) Get entity
            if ( m_entityManager->entityExists(entityName) )
            {
                Ra::Engine::Entity* e = m_entityManager->getEntity(entityName);

                // 2) Get component
                const Ra::Engine::Component* c = e->getComponent(componentName);

                if ( c != nullptr && ! c->m_renderObjects.empty() ) {
                    // 3) Get RO
                    if ( roName.empty() ) {
                        return m_renderObjectManager->getRenderObject(
                                    c->m_renderObjects.front() )->getMesh().get();
                    }
                    else
                    {
                        for (const auto& idx : c->m_renderObjects) {
                            const auto& ro = m_renderObjectManager->getRenderObject(idx);
                            if ( ro->getName() ==  roName) {
                                return ro->getMesh().get();
                            }
                        }
                    }
                }
            }
            return nullptr;
        }
```

**Comment by nmellado** [comment]
Fixed by merged PR #243
